bindings/wallet-core/src/c.rs

```diff
@@ -385,6 +393,107 @@ pub fn wallet_set_state(wallet: WalletPtr, value: u64, counter: u32) -> Result {
     Result::success()
 }
 
+pub unsafe fn wallet_vote_plan(
+    id: *const u8,
+    payload_type: u8,
```

Comment:
this is not a C `enum` on purpose?

Reply:
Hm. Can we do an automatic conversion with `cbindgen`? If so, will do that.

Reply:
depends on what you mean by automatic, I think that if we define an enum then it will get exported as a C enum in the header (which means it won't be a u8, but a... i32?). The conversion from that new type to the current type I believe we need to do manually. I don't know if it's worth it, I just wanted to know if you considered

Reply:
I modified the whole thing to use enums, but I had to define an intermediate enum in `wallet-c`. `cbindgen` could not generate a C enum for `pub use chain_impl_mockchain::vote::PayloadType` even though it has `#[repr(u8)]`.